src/StorageWriter/DAOClaims.ts

```diff
@@ -0,0 +1,62 @@
+import { Claim } from '@po.et/poet-js'
+import { inject, injectable } from 'inversify'
+import { Collection, Db, FindAndModifyWriteOpResultObject } from 'mongodb'
+import { isNil, pipeP, lensPath, view } from 'ramda'
+
+import { NoMoreEntriesException } from './Exceptions'
+
+export interface DAOClaimsConfiguration {
+  readonly maxStorageAttempts: number
+}
+
+const L = {
+  valueClaim: lensPath(['value', 'claim']),
+}
+
+export const getClaimFromFindAndUpdateResponse = (response: FindAndModifyWriteOpResultObject): Claim | undefined =>
+  view(L.valueClaim, response)
+
+export const throwIfClaimNotFound = (claim: Claim): Claim => {
+  if (isNil(claim)) throw new NoMoreEntriesException('No claims found')
+  return claim
+}
+
+@injectable()
+export class DAOClaims {
+  private readonly collection: Collection
+  private readonly maxStorageAttempts: number
+
+  constructor(@inject('DB') db: Db, @inject('DAOClaimsConfiguration') configuration: DAOClaimsConfiguration) {
+    this.collection = db.collection('storageWriterClaims')
+    this.maxStorageAttempts = configuration.maxStorageAttempts
+  }
+
+  public readonly start = async () => {
+    await this.collection.createIndex({ 'claim.id': 1 }, { unique: true })
+  }
+
+  public readonly addClaim = async (claim: Claim) => {
+    await this.collection.insertOne({ claim, storageAttempts: 0, ipfsFileHash: null })
+  }
+
+  public readonly addClaimHash = async (claimId: string, ipfsFileHash: string) => {
+    await this.collection.updateOne({ 'claim.id': claimId }, { $set: { ipfsFileHash } })
+  }
+
+  private readonly findClaimToStore = () =>
+    this.collection.findOneAndUpdate(
```

Comment:
why you omit async/await in this case?

Reply:
typescript infers the type automatically when a promise is returned. The other functions are not returning a value ( on purpose ) so they require async/await.